runtime/caml/memory.h

```diff
@@ -405,6 +405,10 @@ struct caml__roots_block {
     x[caml__i_##x] = Val_unit; \
   }
 
+#define CAMLlocalresult(res) \
+  res.data = Val_unit; \
```

Comment:
shouldn't this be
```suggestion
  caml_result res = Result_unit; \
```
or am I missing something?

Reply:
The `CAMLlocalresult` definition was wrong, and it was not used in the runtime. I believe that it is useful to have such a macro (it is tricky to write, and it is easy to write subtly wrong code by forgetting about it), so what I did was to (1) fix it and (2) tweak the code to use it somewhere (where it is not strictly necessary). As you have seen, #13061 is an offshoot of this work, and I may wait for that one to be resolved to decide what to do here.

I originally introduced `CAMLlocalresult` because I had a use for it in the runtime, but the previous use-case apparently was just in pre-PR experiments and never made it to the PR. It is notable that it is not strictly necessary in the runtime itself, where the cleanup code in the exception case does not require rooting, and the usage code in the value case is either (1)  an immediate return or (2) a continuation assuming that the value is `()` and never using it again.